test/Microsoft.AspNet.Mvc.FunctionalTests/TryValidateModelTest.cs

```diff
@@ -87,5 +81,37 @@ public class TryValidateModelTest
             var body = await response.Content.ReadAsStringAsync();
             Assert.Equal("{}", body);
         }
+
+        [Fact]
+        public async Task TryValidateModel_CollectionsModel_ReturnsErrorsForInvalidProperties()
```

Comment:
Why is this case so special? Should we also add tests for non collection types?
Also I do not see any functional tests for TryUpdateModel, even in that case we should see if we are clearing the dictionary.